cernopendata/modules/pages/views.py

```diff
@@ -190,21 +192,37 @@ def getting_started_redirect(exp):
     """Redirects to associated experiment."""
     if exp == "cms":
         return redirect('/docs/cms-getting-started-2011')
-    return redirect('/docs/%s-getting-started' % exp)
+    elif exp in ['lhcb', 'alice']:  # FIXME to be appended with new docs
+        return redirect('/docs/%s-getting-started' % exp)
+    else:
+        abort(422)
 
 
 @blueprint.route('/vm/<exp>')
 def vm_redirect(exp):
     """Redirects to associated experiment."""
     if exp == "cms":
         return redirect('/docs/cms-virtual-machine-2011')
-    return redirect('/docs/%s-virtual-machine' % exp)
+    elif exp in ['lhcb', 'alice']:  # FIXME to be appended with new docs
+        return redirect('/docs/%s-virtual-machine' % exp)
+    else:
+        abort(422)
+
+
+def check_year(year):
+    """Check if an argument is a year."""
+    if len(year) == 4 and year.isdigit():
+        return True
+    return False
 
 
 @blueprint.route('/vm/<exp>/<year>')
 def vm_redirect_year(exp, year):
     """Redirects to associated experiment."""
-    return redirect('/docs/%s-virtual-machine-%s' % (exp, year), code=302)
+    if check_year(year) and exp in ['lhcb', 'cms', 'alice']:  # FIXME add exp
+        return redirect('/docs/%s-virtual-machine-%s' % (exp, year), code=302)
+    else:
+        abort(422)
```

Comment:
Works well for me, but instead of 422, what about simply returning 404?  Accessing `/vm/xyzzy` is not different from accessing `/foo`; both don't exist.

Reply:
(sorry for the double click 😉 due to slower connection; the comment appplies obviously to all 422 in the patch)